### DISCUSSION TITLE
Support for STM32G0B1RE ?

### DISCUSSION THREAD

**Discussion by Owlmachines** [grblHAL/Controllers#12] [Q&A]
Dear Terje,
Thank you for all your work on GRBL HAL. I am trying to select a board for a CNC Machine. I find the Bigtreetech offering very interesting and I saw there have been successful ports for the SKR mini E3 v2.0 which uses the STM32F103RCT6 processor. I am interested in the Manta E3 EZ board which uses the STM32G0B1RE processor. How likely is it to work out of the box? Or if very unlikely: would there be a lot of work involved in implementing it?  I would be happy to spend some time on it but don't really know where to start. 
Best wishes,
Owl

**Comment by terjeio** [comment]
> How likely is it to work out of the box?

Sadly it will not.

> Or if very unlikely: would there be a lot of work involved in implementing it?

A new driver has to be added, hard to tell how much work it will be - depends on how different it is from the other ST drivers and how much optional functionality to add.

>  I would be happy to spend some time on it but don't really know where to start.

A copy of the F103 driver might be the best to start with since it is a somewhat similar processor and is a pretty basic driver.

**Comment by Owlmachines** [comment]
Dear Terje,
Thanks a lot for your quick reply, very helpful. I might just stick to some already developed options.
Best wishes,
O

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________
